packages/studio-be/src/studio/cloud/cloud-router.ts

```diff
@@ -25,19 +50,103 @@ export class CloudRouter extends CustomStudioRouter {
           scope: 'nlu'
         })
       )
-      .then((res) => `Bearer ${res.data?.access_token}`)
-      .catch((err) => null)
+
+      if (!data.access_token) {
+        throw new UnauthorizedError('Cloud keys are invalid')
+      }
+
+      return `Bearer ${data?.access_token}`
+    } catch (err) {
+      const message = `Error fetching cloud bearer token using cloud config. clientId: ${clientId}, clientSecret: ${
+        clientSecret === undefined ? clientSecret : clientSecret.replace(/./g, '*')
```

Comment:
How can the client token be undefined? Is it just to be extra safe?

```suggestion
        clientSecret?.replace(/./g, '*')
```

Reply:
it was already there. If its undefined I want it to appear "undefined" in the logs so we can know about it